src/ngraph/frontend/onnx_import/op/reshape.cpp

```diff
@@ -0,0 +1,78 @@
+//*****************************************************************************
+// Copyright 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <cstddef>
+#include <memory>
+#include <vector>
+
+#include "ngraph/axis_vector.hpp"
+#include "ngraph/op/constant.hpp"
+#include "ngraph/op/reshape.hpp"
+#include "ngraph/shape.hpp"
+
+#include "exceptions.hpp"
+#include "reshape.hpp"
+#include "utils/reshape.hpp"
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        namespace op
+        {
+            NodeVector reshape(const Node& node)
+            {
+                NodeVector ng_inputs{node.get_ng_inputs()};
+                auto data = ng_inputs.at(0);
+                auto data_shape = data->get_shape();
+
+                auto output_shape = node.get_attribute_value<std::vector<std::size_t>>("shape", {});
+
+                // If no shape argument (opset >= 5) and there is second input.
+                if (output_shape.empty() && ng_inputs.size() == 2)
+                {
+                    auto output_shape_node =
```

Comment:
Don't get this line. Why we need `std::dynamic_pointer_cast<>`? You expect `ng_input.at(1)` to be `nullptr`??? It's impossible. In other words, if this is truly `nullptr` then this is a bug, not a feature.